### PR TITLE
Gb/nc chunks

### PR DESCRIPTION
**Pull request by grantbuster**

### CODE REVIEW

**Review by bnb32 — COMMENTED** [review]
Looks good. Just a couple questions/comments.

File: sup3r/preprocessing/data_handling.py [hunk 1]
```diff
@@ -1748,10 +1748,113 @@ def get_raster_index(self):
...
+                                    .format(feature, os.path.basename(fp)))
+                        self.data[..., idf] *= scalar
+                        self.data[..., idf] += adder
+                        completed.append(feature)
+
 
 class DataHandlerNC(DataHandler):
     """Data Handler for NETCDF data"""
 
+    CHUNKS = {'XTIME': 100, 'XLAT': 100, 'XLON': 100}
```
[the remark this answers is not in the document]
Comment (grantbuster) [reply sup3r/preprocessing/data_handling.py:1834]: Yep agreed, do they have those names exactly? `south_north` and `west_east`? Any others we should add?

File: sup3r/utilities/utilities.py [hunk 2]
```diff
@@ -989,15 +999,27 @@ def interp_to_level(var_array, lev_array, levels):
     shape = (len(levels), array_shape[-4], np.product(array_shape[-2:]))
     out_array = np.zeros(shape, dtype=np.float32).T
 
-    for i in range(array_shape[0]):
+    # if multiple vertical levels have identical heights at the desired
+    # interpolation level, interpolation to that value will fail because linear
+    # slope will be NaN. This is most common if you have multiple pressure
+    # levels at zero height at the surface in the case that the data didnt
+    # provide underground data.
+    for level in levels:
```
[the remark this answers is not in the document]
Comment (grantbuster) [reply sup3r/utilities/utilities.py:1007]: Correct, for example if the data has a floor geopotential height of 0m, multiple vertical levels are at 0m, and you're interpolating to 0m, the interpolation will throw an error. Adding a bit of random noise to any levels that are exactly at the target interpolation level is an easy fix.

File: sup3r/preprocessing/data_handling.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (grantbuster) [reply sup3r/preprocessing/data_handling.py:1834]: Also do you want to change the chunk sizes at all?

[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/preprocessing/data_handling.py:1834]: Yeah those are the exact names. I'm currently using 120x120 for fwp_shape so maybe we could put the default somewhere above that. 150x150?

File: sup3r/utilities/utilities.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/utilities/utilities.py:1007]: Yeah that makes sense. Is there any weirdness that might come from this noise?

File: sup3r/preprocessing/data_handling.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/preprocessing/data_handling.py:1834]: We might also want to add "bottom_top" for the pressure levels and "Time".

File: sup3r/utilities/utilities.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (grantbuster) [reply sup3r/utilities/utilities.py:1007]: Not really, adding 1e-5 to the x values in a lin interp shouldn't really change anything

[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/utilities/utilities.py:1007]: Yeah that was my impression as well

[the remark this answers is not in the document]
Comment (grantbuster) [reply sup3r/utilities/utilities.py:1007]: I should also add that this is incredibly rare - you should basically never get vertical levels exactly at the value you're interpolating to. This really only happens when we fix a floor at surface.

[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/utilities/utilities.py:1007]: Yeah good point

**Review by bnb32 — APPROVED** [review]

**Pull request merged**